### PR TITLE
Add algorithm that tests knowledge of strings

### PR DESCRIPTION
**Pull request by SaadBenn**

### CODE REVIEW

File: strings/validate_coordinates.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        longitude = float(l[1])
+    except:
+        return False
+    print(latitude, longitude)
+    return -90 <= latitude <= 90 and -180 <= longitude <= 180
+
+# friends solutions
+def is_valid_coordinates_1(coordinates):
+    try:
+        lat, lng = [abs(float(c)) for c in coordinates.split(',') if 'e' not in c]
```
Comment (christianbender) [inline strings/validate_coordinates.py]: What means ```'e'``` ?

File: strings/validate_coordinates.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+And some invalid ones:
+23.234, - 23.4234
+N23.43345, E32.6457
+6.325624, 43.34345.345
+0, 1,2
+
+"""
+# I'll be adding my attempt as well as my friend's solution (took us ~ 1 hour)
+
```
Comment (christianbender) [inline strings/validate_coordinates.py]: Add ```import unittest```

File: strings/validate_coordinates.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	
+    def test_valid(self):
+    	valid_coordinates = [
+    		"-23, 25",
+    		"4, -3",
+    		"90, 180",
+    		"-90, -180"]
+
+    	for coordinate in valid_coordinates:
+    		self.assertTrue(is_valid_coordinates(coordinate))
```
Comment (christianbender) [inline strings/validate_coordinates.py]: The function ```is_valid_coordinates(...)``` don't exists.

File: strings/validate_coordinates.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    def test_invalid(self):
+    	invalid_coordinates = [
+		"23.234, - 23.4234",
+		"99.234, 12.324",
+		"6.325624, 43.34345.345",
+		"0, 1,2",
+		"23.245, 1e1"]
+
+		for coordinate in invalid_coordinates:
+    		self.assertFalse(is_valid_coordinates(coordinate))
```
Comment (christianbender) [inline strings/validate_coordinates.py]: The same thing as above.

File: strings/validate_coordinates.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+def is_valid_coordinates_1(coordinates):
+    try:
+        lat, lng = [abs(float(c)) for c in coordinates.split(',') if 'e' not in c]
+    except ValueError:
+        return False
+
+    return lat <= 90 and lng <= 180
+
+# using regular expression
+import re
```
Comment (christianbender) [inline strings/validate_coordinates.py]: Put this import to the begin of the file.

File: strings/validate_coordinates.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+def is_valid_coordinates_regular_expression(coordinates):
+    return bool(re.match("-?(\d|[1-8]\d|90)\.?\d*, -?(\d|[1-9]\d|1[0-7]\d|180)\.?\d*$", coordinates))
```
Comment (christianbender) [inline strings/validate_coordinates.py]: Please break up this line. max. 80 columns. if possible

File: strings/validate_coordinates.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+N23.43345, E32.6457
+6.325624, 43.34345.345
+0, 1,2
+
+"""
+# I'll be adding my attempt as well as my friend's solution (took us ~ 1 hour)
+
+# my attempt
+def is_valid_coordinates_0(coordinates):
+    print(coordinates)
```
Comment (goswami-rahul) [inline strings/validate_coordinates.py]: no need for `print` statements.

File: strings/validate_coordinates.py (hunk 1 above)
Comment (goswami-rahul) [reply strings/validate_coordinates.py]: I think he means to check for numbers with exponents like `12e5`. Or I may be completely wrong.

**Comment by goswami-rahul** [comment]
@SaadBenn I see that you have created this branch in this original repo. That is OK, but it is preferred to maintain your changes in your own fork's branches, and then pull them here.

File: strings/validate_coordinates.py (hunk 1 above)
Comment (SaadBenn) [reply strings/validate_coordinates.py]: It checks for exponents.

File: strings/validate_coordinates.py (hunk 5 above)
Comment (SaadBenn) [reply strings/validate_coordinates.py]: I think It is always a good idea to limit the scope of import statements and variables and define them where they are needed. But If this is the requirement of this repo then I don't have problem changing it. @goswami-rahul

File: strings/validate_coordinates.py (hunk 6 above)
Comment (SaadBenn) [reply strings/validate_coordinates.py]: This is a regular expression. Breaking up a statement like this is very illogical, IMHO.

**Comment by SaadBenn** [comment]
I'll heed to that in future. I'll delete this branch as soon as the merge is complete. @goswami-rahul

**Pull request merged**